src/cace/cace_ari.c

```diff
@@ -113,13 +113,20 @@ static int read_text(ari_t *inval, FILE *source)
     free(buf);
 
     const char *errm = NULL;
-    res              = ari_text_decode(inval, intext, &errm);
+#if defined(ARI_TEXT_PARSE)
```

Comment:
This leaks memory from `intext` because of this condition. The conditional part could be the entire function body because reading the line will not be useful if the decoder cannot work.

It may be better to actually disallow the in-form to be text entirely, which would avoid the calls to `read_text()`. We can adjust the CACE library API to provide an always-failing `ari_text_decode()` if it makes the compiler logic easier here.

Reply:
Good point. Refactored to completely omit `read_text` and `read_auto` if `ARI_TEXT_PARSE` is not defined.